gateway/coprocess.go

```diff
@@ -193,8 +193,12 @@ func (c *CoProcessor) ObjectPostProcess(object *coprocess.Object, r *http.Reques
 		values.Set(p, v)
 	}
 
-	r.URL, _ = url.ParseRequestURI(object.Request.Url)
+	r.URL, err = url.ParseRequestURI(object.Request.Url)
+	if err != nil {
+		return
```

Comment:
@matiasinsaurralde don't we want to print this error or return it?

Reply:
@matiasinsaurralde consider using Parse which calls internal parse with `viaRequest` set to false

Reply:
Also maybe we should add a test (since we don't check if `r.url == nil` and we trust that function that if err is nil then the pointer isn't nil, but someone might change it at some point, maybe to allow empty string or nil as an input etc.)

Reply:
Last thing (I hope :) ) - could it be that the right fix would be to change the function to use Parse?  https://golang.org/src/net/url/url.go line 472 since the request to the plugin can come from anywhere so maybe it's too much to assume, and we should allow this use case?

Reply:
@letzya About printing/returning the error, with this change, it will achieve that.
The first call is `ProcessRequest`, which dispatches the object and calls `ObjectPostProcess` with the modified request object. In case `ObjectPostProcess` fails, we will see the error!